source/simulator/solver.cc

```diff
@@ -253,25 +253,92 @@ namespace aspect
       S_block_tolerance(S_block_tolerance)
     {}
 
-    template <class PreconditionerA, class PreconditionerMp>
+    template <class PreconditionerA, class SInvOperator>
     unsigned int
-    BlockSchurPreconditioner<PreconditionerA, PreconditionerMp>::
+    BlockSchurPreconditioner<PreconditionerA, SInvOperator>::
     n_iterations_A() const
     {
       return n_iterations_A_;
     }
 
-    template <class PreconditionerA, class PreconditionerMp>
+    template <class PreconditionerA, class SInvOperator>
     unsigned int
-    BlockSchurPreconditioner<PreconditionerA, PreconditionerMp>::
+    BlockSchurPreconditioner<PreconditionerA, SInvOperator>::
     n_iterations_S() const
     {
-      return n_iterations_S_;
+      return S_inv_operator.n_iterations;
     }
 
-    template <class PreconditionerA, class PreconditionerMp>
+    template <class PreconditionerMp>
+
+    //Used to implement right preconditioner in which
```

Comment:
Can you clarify this comment a bit? I am not quite sure what the `Used to` part means. Do you mean "it used to do this in the past", or "this class is used to do ...". Maybe you mean: `This class is used in the implementation of the right preconditioner. It approximates that Schur complement as the pressure mass matrix weighted by the inverse of the viscosity.`

Also just for readability: Please put spaces between the backslash and the start of the comment, I think that is our usual convention for comments.

Reply:
Actually, for class-level documentation, you also want to use the `/** ... */` style.